ocaml/xapi-storage-script/main.ml

```diff
@@ -313,25 +605,13 @@ let _is_a_snapshot_key = "is_a_snapshot"
 
 let _snapshot_of_key = "snapshot_of"
 
-let is_executable path =
-  Async.Sys.is_file ~follow_symlinks:true path >>= function
-  | `No | `Unknown ->
-      return (Error (`missing path))
-  | `Yes -> (
-      Async.Unix.access path [`Exec] >>= function
-      | Error exn ->
-          return (Error (`not_executable (path, exn)))
-      | Ok () ->
-          return (Ok ())
-    )
-
 module Script = struct
   (** We cache (lowercase script name -> original script name) mapping for the
       scripts in the root directory of every registered plugin. *)
   let name_mapping = Base.Hashtbl.create ~size:4 (module Base.String)
 
   let update_mapping ~script_dir =
-    Async.Sys.readdir script_dir >>| Array.to_list >>| fun files ->
+    Sys.readdir script_dir >>= fun files ->
```

Comment:
This is slightly confusing because there is a `Sys.readdir` in OCaml's stdlib, which this one shadows.
Although it is a useful way to avoid accidentally calling the blocking one.